classes/manager.php

```diff
@@ -1536,4 +1533,15 @@ private function mark_push_as_failed(
         // Add failed transfer log.
         $this->save_transfer_log($requestidentifier, $assessmentmappingid, $userid, null, $response, $taskid, intval($errorlogid));
     }
+
+    /**
+     * Get a list of the allowed course modules for grade push.
+     * @return string[] e.g. ['assign', 'quiz', 'turnitintooltwo']
+     */
+    public static function allowed_activities(): array {
+        $mods = \core\plugininfo\mod::get_enabled_plugins();
+        return array_values(array_filter($mods, function($mod): bool {
```

Comment:
this will return all mods enabled in Moodle including a bunch we really don't expect marks from. 

maybe we can have site level setting to have them allowed / excluded?

Reply:
It won't show all the enabled activities as it will check if the associated class is exists in the next line. (ln 1544).
But I agree site level setting is a better approach.

Reply:
I saw the thing about checking if the class exists but didn’t see any new classes being added so I assumed it was trying to add support for all mods in clever way